### DISCUSSION TITLE
Is the preprocess of test_data not allowed by our model?

### DISCUSSION THREAD

**Discussion by yuhao-morning** [PaddlePaddle/PaddleSpatial#98] [BaiduKDDCup2022] [answered]
From the github https://github.com/PaddlePaddle/PaddleSpatial/blob/main/apps/wpf_baseline_gru/kddcup22-sdwpf-evaluation , I see that we only upload predict.py, prepare.py.  The evaluation.py in this folder only executes our predict.py to call forecast() method, and all preprocess of test_data is fixed in the backend and written in test_data.py and evaluation.py. Does that mean we can't do any preprocess of test_data ? I was confused.

**Comment by ramber1836** [comment, marked as answer]
You can preprocess the input of the test data, i.e. test_x. 
As test_y is the ground truth and cannot be accessed by the competitors.

**Comment by yuhao-morning** [comment]
Do you mean we need preprocess test_x locally, and then replace the original one with the preprocessed one in hand?

 Whe can't we preprocess the test_x in code (maybe in predict.py) ?













At 2022-05-24 17:49:47, "Xinjiang Lu" ***@***.***> wrote:

You can preprocess the input of the test data, i.e. test_x.
As test_y is the ground truth and cannot be accessed by the competitors.

**Comment by ramber1836** [comment]
What I mean is that test_x CAN be processed by you.
